lib/hooks/useGetOrgRepos.ts

```diff
@@ -0,0 +1,59 @@
+import useSWR, { Fetcher } from "swr";
+import { supabase } from "lib/utils/supabase";
+
+const baseUrl = "https://api.github.com";
+
+const githubApiRepoFetcher: Fetcher = async (apiUrl: string) => {
+  const sessionResponse = await supabase.auth.getSession();
+  const sessionToken = sessionResponse?.data.session?.provider_token;
+  const defaultHeaders = {
+    accept: "application/json",
+  };
+  const res = await fetch(`${baseUrl}/${apiUrl}`, {
+    headers: {
+      ...defaultHeaders,
+      ...(sessionToken
+        ? {
+            Authorization: `Bearer ${sessionToken}`,
+          }
+        : {}),
+    },
+  });
+
+  if (!res.ok) {
+    const error = new Error("HttpError");
+
+    error.message = `${res.status} ${res.statusText}`;
+    error.stack = JSON.stringify(await res.json());
+    // eslint-disable-next-line no-console
+    console.error(error);
+
+    throw error;
+  }
+
+  return res.json();
+};
+
+export const useGetOrgRepos = ({ organization, limit = 100 }: { organization: string | undefined; limit?: number }) => {
```

Comment:
TODO: get all pages of data. See https://docs.github.com/en/rest/using-the-rest-api/using-pagination-in-the-rest-api?apiVersion=2022-11-28

Reply:
I ran into this soft limit with some of the bigger orgs I'm in:

![Screenshot 2024-02-12 at 8 22 08 AM](https://github.com/open-sauced/app/assets/23109390/9c69e75c-bf1e-4fbd-93ee-d9514f256e7f)

only 100 get loaded but we can fix that in a subsequent PR! This is great!

Reply:
> I ran into this soft limit with some of the bigger orgs I'm in:
> 
> ![Screenshot 2024-02-12 at 8 22 08 AM](https://private-user-images.githubusercontent.com/23109390/304135318-9c69e75c-bf1e-4fbd-93ee-d9514f256e7f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc3NTE5MjgsIm5iZiI6MTcwNzc1MTYyOCwicGF0aCI6Ii8yMzEwOTM5MC8zMDQxMzUzMTgtOWM2OWU3NWMtYmYxZS00ZmJkLTkzZWUtZDk1MTRmMjU2ZTdmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjEyVDE1MjcwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkNDFlOWM2ZGI0ZTI4ZjQwMDE4OWUyYTExNmFjNjBmN2E0N2QyYWEzNDg3OGRmYjI4NWI4OGJlNWU0OTJkMjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.nFzH3hGWRNWsabZ_SZTNUIG-aKEnVkmkkLQ6Tos0b04)
> 
> only 100 get loaded but we can fix that in a subsequent PR! This is great!

There is paged data that needs to be pulled in. See https://github.com/open-sauced/app/pull/2630/files#r1484876958 . I was planning to tackle that in a follow-up PR.

Reply:
Followup PR is perfect!

Reply:
Follow up PR in #2650.